action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -406,6 +416,17 @@ func (stateDB *StateDBAdapter) Suicide(evmAddr common.Address) bool {
 		stateDB.logError(err)
 		return false
 	}
+	// before calling Suicide, EVM will transfer the contract's balance to beneficiary
+	// need to create a transaction log on successful suicide
+	if len(stateDB.lastAddBalanceAmount.Bytes()) > 0 {
+		from, _ := address.FromBytes(evmAddr[:])
+		stateDB.addTransactionLogs(&action.TransactionLog{
```

Comment:
if lastAddBalanceAmount is not set or the amount is different from balance, panic?

Reply:
fixed in latest commit